src/test/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsFormatTests.java

```diff
@@ -0,0 +1,209 @@
+        // ensuring segments are created
+        indexWriter.flush();
+        indexWriter.commit();
+        indexWriter.close();
+
+        // Validate to see if correct values are returned, assumption here is only 1 segment is getting created
+        IndexSearcher searcher = new IndexSearcher(indexReader);
+        final LeafReader leafReader = searcher.getLeafContexts().get(0).reader();
+        SegmentReader segmentReader = Lucene.segmentReader(leafReader);
+        final List<String> hnswfiles = getFilesFromSegment(dir, HNSW_FILE_EXT);
+        // 0 hnsw files for now as we have not integrated graph creation here.
+        assertEquals(0, hnswfiles.size());
+        assertEquals(hnswfiles.stream().filter(x -> x.contains(FLOAT_VECTOR_FIELD)).count(), 0);
+        assertEquals(hnswfiles.stream().filter(x -> x.contains(BYTE_VECTOR_FIELD)).count(), 0);
+
+        // Even setting IWC to not use compound file it still uses compound file, hence ensuring we don't check .vec
+        // file in case segment uses compound format. use this seed once we fix this to validate everything is
+        // working or not. -Dtests.seed=CAAE1B8D573EEB7E
+        if (segmentReader.getSegmentInfo().info.getUseCompoundFile() == false) {
+            final List<String> vecfiles = getFilesFromSegment(dir, FLAT_VECTOR_FILE_EXT);
+            // 2 .vec files will be created as we are using per field vectors format.
+            assertEquals(2, vecfiles.size());
+        }
+
+        final FloatVectorValues floatVectorValues = leafReader.getFloatVectorValues(FLOAT_VECTOR_FIELD);
+        floatVectorValues.nextDoc();
+        assertArrayEquals(floatVector, floatVectorValues.vectorValue(), 0.0f);
+        assertEquals(1, floatVectorValues.size());
+        assertEquals(3, floatVectorValues.dimension());
+
+        final ByteVectorValues byteVectorValues = leafReader.getByteVectorValues(BYTE_VECTOR_FIELD);
+        byteVectorValues.nextDoc();
+        assertArrayEquals(byteVector, byteVectorValues.vectorValue());
+        assertEquals(1, byteVectorValues.size());
+        assertEquals(2, byteVectorValues.dimension());
+
+        Assert.assertThrows(
+            UnsupportedOperationException.class,
+            () -> leafReader.searchNearestVectors(FLOAT_VECTOR_FIELD, floatVector, 10, new Bits.MatchAllBits(1), 10)
+        );
+
+        Assert.assertThrows(
+            UnsupportedOperationException.class,
+            () -> leafReader.searchNearestVectors(BYTE_VECTOR_FIELD, byteVector, 10, new Bits.MatchAllBits(1), 10)
+        );
+        // do it at the end so that all search is completed
+        indexReader.close();
+    }
+
+    private List<String> getFilesFromSegment(Directory dir, String fileFormat) throws IOException {
+        return Arrays.stream(dir.listAll()).filter(x -> x.contains(fileFormat)).collect(Collectors.toList());
+    }
+
+    /**
+     * This should have been annotated with @Before, but somehow when I annotate with @Before apart from running
```

Comment:
Isn't it because you have two tests in this class? Then, this will be called twice for each test.
If it runs more than that, it could be due to method name is being conflicted where it is being called by parent class?

Reply:
Could be. the problem was when I run it there is some test called as classMethod which was getting called and I was not able to figure that at the time of raising the PR. I will debug this more but as it is in test this was not a big issue. Hence I kept it. But I understand your concern let me try to give it 1 more shot to see if I can fix the problem.